src/cdk/table/coalesced-style-scheduler.ts

```diff
@@ -62,19 +62,31 @@ export class _CoalescedStyleScheduler implements OnDestroy {
 
     this._currentSchedule = new _Schedule();
 
-    this._ngZone.onStable.pipe(
-        take(1),
+    this._getScheduleObservable().pipe(
         takeUntil(this._destroyed),
     ).subscribe(() => {
-      const schedule = this._currentSchedule!;
-      this._currentSchedule = null;
+      while (this._currentSchedule!.tasks.length || this._currentSchedule!.endTasks.length) {
+        const schedule = this._currentSchedule!;
+        this._currentSchedule = new _Schedule();
```

Comment:
Add a comment that explains why you assign a new schedule and then set back to `null` at the end of the loop?